4.x: DefaultMetadataTabletMapIT leaves multiple hanging sessions

Observed (but not only) for example in:
https://github.com/scylladb/java-driver/actions/runs/18193758711/job/51794652116
where it causes AdvancedShardAwarenessIT to fail with:
```
Error:  Failures: 
Error:    AdvancedShardAwarenessIT.should_initialize_all_channels:133->lambda$should_initialize_all_channels$2:133->assertNoLogMatches:276 Logs should not contain pattern [.*Scheduling next reconnection in.*] but found in [[s92|/127.0.18.1:9042] Scheduling next reconnection in PT1M]
Error:    AdvancedShardAwarenessIT.should_initialize_all_channels:133->lambda$should_initialize_all_channels$2:133->assertNoLogMatches:276 Logs should not contain pattern [.*Scheduling next reconnection in.*] but found in [[s72] Scheduling next reconnection in PT1M]
Error:    AdvancedShardAwarenessIT.should_not_struggle_to_fill_pools:248->assertMatchesAtMost:331 Expected to find pattern at most 4 times but found it 5 times. Pattern: [.*Scheduling next reconnection in.*]
```
However this cluster can be traced back to
```
[INFO] Running com.datastax.oss.driver.core.metadata.RandomTokenVnodesIT
13:10:08.614 [main] INFO  c.d.o.d.api.testinfra.ccm.CcmBridge - Executing: [ccm, create, ccm_1, -i, 127.0.18., -n, 2:0, -v, release:6.2.3, --scylla, --config-dir=/tmp/ccm3788213650875537416]
13:10:08.960 [Exec Stream Pumper] INFO  c.d.o.d.api.testinfra.ccm.CcmBridge - ccmout> Current cluster is now: ccm_1
13:10:08.993 [main] INFO  c.d.o.d.api.testinfra.ccm.CcmBridge - Executing: [ccm, updateconf, auto_snapshot:false, experimental_features:[consistent-topology-changes,tablets], --config-dir=/tmp/ccm3788213650875537416]
13:10:09.304 [main] INFO  c.d.o.d.api.testinfra.ccm.CcmBridge - Executing: [ccm, start, --wait-for-binary-proto, --wait-other-notice, --config-dir=/tmp/ccm3788213650875537416]
Warning:  Tests run: 1, Failures: 0, Errors: 0, Skipped: 1, Time elapsed: 6.87 s - in com.datastax.oss.driver.core.metadata.RandomTokenVnodesIT
[INFO] Running com.datastax.oss.driver.core.metadata.DefaultMetadataTabletMapIT
13:10:24.703 [main] INFO  c.d.o.d.api.testinfra.ccm.CcmBridge - Executing: [ccm, remove, --config-dir=/tmp/ccm3788213650875537416]
[INFO] Tests run: 2, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 10.529 s - in com.datastax.oss.driver.core.metadata.DefaultMetadataTabletMapIT
[INFO] Running com.datastax.oss.driver.core.metadata.ByteOrderedTokenIT
Warning:  Tests run: 1, Failures: 0, Errors: 0, Skipped: 1, Time elapsed: 0.007 s - in com.datastax.oss.driver.core.metadata.ByteOrderedTokenIT
```
RandomTokenVnodesIT is currently marked with annotations that should make it irrelevant for Scylla anyway:
https://github.com/scylladb/java-driver/blob/924a28aa17a96579f695bb47accbf8ee3ae0725b/integration-tests/src/test/java/com/datastax/oss/driver/core/metadata/RandomTokenVnodesIT.java#L41-L47

Since the failures started appearing it started taking around 6.8 seconds rather than no more than 1.
Even though the placement of log lines about creation of cluster (127.0.18.) make it almost look like it's created for RandomTokenVnodesIT i think it actually is created for the next test in the logs, which is DefaultMetadataTabletMapIT. The updateconf command matches the requirements for it, while it not matches the requirements for RandomTokenVnodesIT.
The number of nodes is also 2 rather than 3.

https://github.com/scylladb/java-driver/blob/965c9a70382cf96fec73f74f7c6ea883d719db57/integration-tests/src/test/java/com/datastax/oss/driver/core/metadata/DefaultMetadataTabletMapIT.java#L45-L52

Throughout the job it seems that the session attempts to communicate with shutdown cluster for several minutes, until it causes different test to fail.
```

01:34:47.635 [s77-admin-0] DEBUG c.d.o.d.i.core.pool.ChannelPool - [s77|/127.0.18.1:9042] Reconnection attempt complete, 0/2 channels
01:34:47.635 [s77-admin-0] DEBUG c.d.o.d.i.c.u.c.Reconnection - [s77|/127.0.18.1:9042] Scheduling next reconnection in PT58.2S
01:34:48.265 [s68-admin-1] DEBUG c.d.o.d.i.core.pool.ChannelPool - [s68|/127.0.18.2:9042] Trying to create 2 missing channels
01:34:48.267 [s68-admin-1] WARN  c.d.o.d.i.core.pool.ChannelPool - [s68|/127.0.18.2:9042]  Error while opening new channel
com.datastax.oss.driver.api.core.connection.ConnectionInitException: [s68|connecting...] Protocol initialization request, step 1 (OPTIONS): failed to send request (io.netty.channel.StacklessClosedChannelException)
	at com.datastax.oss.driver.internal.core.channel.ProtocolInitHandler$InitRequest.fail(ProtocolInitHandler.java:375)
	at com.datastax.oss.driver.internal.core.channel.ChannelHandlerRequest.writeListener(ChannelHandlerRequest.java:89)
	at io.netty.util.concurrent.DefaultPromise.notifyListener0(DefaultPromise.java:590)
	at io.netty.util.concurrent.DefaultPromise.notifyListenersNow(DefaultPromise.java:557)
	at io.netty.util.concurrent.DefaultPromise.notifyListeners(DefaultPromise.java:492)
	at io.netty.util.concurrent.DefaultPromise.addListener(DefaultPromise.java:185)
	at io.netty.channel.DefaultChannelPromise.addListener(DefaultChannelPromise.java:95)
	at io.netty.channel.DefaultChannelPromise.addListener(DefaultChannelPromise.java:30)
	at com.datastax.oss.driver.internal.core.channel.ChannelHandlerRequest.send(ChannelHandlerRequest.java:78)
	at com.datastax.oss.driver.internal.core.channel.ProtocolInitHandler$InitRequest.send(ProtocolInitHandler.java:209)
	at com.datastax.oss.driver.internal.core.channel.ProtocolInitHandler.onRealConnect(ProtocolInitHandler.java:136)
	at com.datastax.oss.driver.internal.core.channel.ConnectInitHandler.lambda$connect$0(ConnectInitHandler.java:59)
	at io.netty.util.concurrent.DefaultPromise.notifyListener0(DefaultPromise.java:590)
	at io.netty.util.concurrent.DefaultPromise.notifyListeners0(DefaultPromise.java:583)
	at io.netty.util.concurrent.DefaultPromise.notifyListenersNow(DefaultPromise.java:559)
	at io.netty.util.concurrent.DefaultPromise.notifyListeners(DefaultPromise.java:492)
	at io.netty.util.concurrent.DefaultPromise.setValue0(DefaultPromise.java:636)
	at io.netty.util.concurrent.DefaultPromise.setFailure0(DefaultPromise.java:629)
	at io.netty.util.concurrent.DefaultPromise.tryFailure(DefaultPromise.java:118)
	at io.netty.channel.nio.AbstractNioChannel$AbstractNioUnsafe.fulfillConnectPromise(AbstractNioChannel.java:326)
	at io.netty.channel.nio.AbstractNioChannel$AbstractNioUnsafe.finishConnect(AbstractNioChannel.java:342)
	at io.netty.channel.nio.NioEventLoop.processSelectedKey(NioEventLoop.java:784)
	at io.netty.channel.nio.NioEventLoop.processSelectedKeysOptimized(NioEventLoop.java:732)
	at io.netty.channel.nio.NioEventLoop.processSelectedKeys(NioEventLoop.java:658)
	at io.netty.channel.nio.NioEventLoop.run(NioEventLoop.java:562)
	at io.netty.util.concurrent.SingleThreadEventExecutor$4.run(SingleThreadEventExecutor.java:998)
	at io.netty.util.internal.ThreadExecutorMap$2.run(ThreadExecutorMap.java:74)
	at io.netty.util.concurrent.FastThreadLocalRunnable.run(FastThreadLocalRunnable.java:30)
	at java.lang.Thread.run(Thread.java:750)
	Suppressed: io.netty.channel.AbstractChannel$AnnotatedConnectException: Connection refused: /127.0.18.2:9042
	Caused by: java.net.ConnectException: Connection refused
		at sun.nio.ch.SocketChannelImpl.checkConnect(Native Method)
		at sun.nio.ch.SocketChannelImpl.finishConnect(SocketChannelImpl.java:716)
		at io.netty.channel.socket.nio.NioSocketChannel.doFinishConnect(NioSocketChannel.java:336)
		at io.netty.channel.nio.AbstractNioChannel$AbstractNioUnsafe.finishConnect(AbstractNioChannel.java:339)
		at io.netty.channel.nio.NioEventLoop.processSelectedKey(NioEventLoop.java:784)
		at io.netty.channel.nio.NioEventLoop.processSelectedKeysOptimized(NioEventLoop.java:732)
		at io.netty.channel.nio.NioEventLoop.processSelectedKeys(NioEventLoop.java:658)
		at io.netty.channel.nio.NioEventLoop.run(NioEventLoop.java:562)
		at io.netty.util.concurrent.SingleThreadEventExecutor$4.run(SingleThreadEventExecutor.java:998)
		at io.netty.util.internal.ThreadExecutorMap$2.run(ThreadExecutorMap.java:74)
		at io.netty.util.concurrent.FastThreadLocalRunnable.run(FastThreadLocalRunnable.java:30)
		at java.lang.Thread.run(Thread.java:750)
Caused by: io.netty.channel.StacklessClosedChannelException: null
	at io.netty.channel.AbstractChannel$AbstractUnsafe.flush0()(Unknown Source)
01:34:48.269 [s68-admin-1] WARN  c.d.o.d.i.core.pool.ChannelPool - [s68|/127.0.18.2:9042]  Error while opening new channel
com.datastax.oss.driver.api.core.connection.ConnectionInitException: [s68|connecting...] Protocol initialization request, step 1 (OPTIONS): failed to send request (io.netty.channel.StacklessClosedChannelException)
	at com.datastax.oss.driver.internal.core.channel.ProtocolInitHandler$InitRequest.fail(ProtocolInitHandler.java:375)
	at com.datastax.oss.driver.internal.core.channel.ChannelHandlerRequest.writeListener(ChannelHandlerRequest.java:89)
	at io.netty.util.concurrent.DefaultPromise.notifyListener0(DefaultPromise.java:590)
	at io.netty.util.concurrent.DefaultPromise.notifyListenersNow(DefaultPromise.java:557)
	at io.netty.util.concurrent.DefaultPromise.notifyListeners(DefaultPromise.java:492)
	at io.netty.util.concurrent.DefaultPromise.addListener(DefaultPromise.java:185)
	at io.netty.channel.DefaultChannelPromise.addListener(DefaultChannelPromise.java:95)
	at io.netty.channel.DefaultChannelPromise.addListener(DefaultChannelPromise.java:30)
	at com.datastax.oss.driver.internal.core.channel.ChannelHandlerRequest.send(ChannelHandlerRequest.java:78)
	at com.datastax.oss.driver.internal.core.channel.ProtocolInitHandler$InitRequest.send(ProtocolInitHandler.java:209)
	at com.datastax.oss.driver.internal.core.channel.ProtocolInitHandler.onRealConnect(ProtocolInitHandler.java:136)
	at com.datastax.oss.driver.internal.core.channel.ConnectInitHandler.lambda$connect$0(ConnectInitHandler.java:59)
	at io.netty.util.concurrent.DefaultPromise.notifyListener0(DefaultPromise.java:590)
	at io.netty.util.concurrent.DefaultPromise.notifyListeners0(DefaultPromise.java:583)
	at io.netty.util.concurrent.DefaultPromise.notifyListenersNow(DefaultPromise.java:559)
	at io.netty.util.concurrent.DefaultPromise.notifyListeners(DefaultPromise.java:492)
	at io.netty.util.concurrent.DefaultPromise.setValue0(DefaultPromise.java:636)
	at io.netty.util.concurrent.DefaultPromise.setFailure0(DefaultPromise.java:629)
	at io.netty.util.concurrent.DefaultPromise.tryFailure(DefaultPromise.java:118)
	at io.netty.channel.nio.AbstractNioChannel$AbstractNioUnsafe.fulfillConnectPromise(AbstractNioChannel.java:326)
	at io.netty.channel.nio.AbstractNioChannel$AbstractNioUnsafe.finishConnect(AbstractNioChannel.java:342)
	at io.netty.channel.nio.NioEventLoop.processSelectedKey(NioEventLoop.java:784)
	at io.netty.channel.nio.NioEventLoop.processSelectedKeysOptimized(NioEventLoop.java:732)
	at io.netty.channel.nio.NioEventLoop.processSelectedKeys(NioEventLoop.java:658)
	at io.netty.channel.nio.NioEventLoop.run(NioEventLoop.java:562)
	at io.netty.util.concurrent.SingleThreadEventExecutor$4.run(SingleThreadEventExecutor.java:998)
	at io.netty.util.internal.ThreadExecutorMap$2.run(ThreadExecutorMap.java:74)
	at io.netty.util.concurrent.FastThreadLocalRunnable.run(FastThreadLocalRunnable.java:30)
	at java.lang.Thread.run(Thread.java:750)
	Suppressed: io.netty.channel.AbstractChannel$AnnotatedConnectException: Connection refused: /127.0.18.2:9042
	Caused by: java.net.ConnectException: Connection refused
		at sun.nio.ch.SocketChannelImpl.checkConnect(Native Method)
		at sun.nio.ch.SocketChannelImpl.finishConnect(SocketChannelImpl.java:716)
		at io.netty.channel.socket.nio.NioSocketChannel.doFinishConnect(NioSocketChannel.java:336)
		at io.netty.channel.nio.AbstractNioChannel$AbstractNioUnsafe.finishConnect(AbstractNioChannel.java:339)
		at io.netty.channel.nio.NioEventLoop.processSelectedKey(NioEventLoop.java:784)
		at io.netty.channel.nio.NioEventLoop.processSelectedKeysOptimized(NioEventLoop.java:732)
		at io.netty.channel.nio.NioEventLoop.processSelectedKeys(NioEventLoop.java:658)
		at io.netty.channel.nio.NioEventLoop.run(NioEventLoop.java:562)
		at io.netty.util.concurrent.SingleThreadEventExecutor$4.run(SingleThreadEventExecutor.java:998)
		at io.netty.util.internal.ThreadExecutorMap$2.run(ThreadExecutorMap.java:74)
		at io.netty.util.concurrent.FastThreadLocalRunnable.run(FastThreadLocalRunnable.java:30)
		at java.lang.Thread.run(Thread.java:750)
Caused by: io.netty.channel.StacklessClosedChannelException: null
	at io.netty.channel.AbstractChannel$AbstractUnsafe.flush0()(Unknown Source)
01:34:48.269 [s68-admin-1] DEBUG c.d.o.d.i.core.pool.ChannelPool - [s68|/127.0.18.2:9042] Reconnection attempt complete, 0/2 channels
01:34:48.269 [s68-admin-1] DEBUG c.d.o.d.i.c.u.c.Reconnection - [s68|/127.0.18.2:9042] Scheduling next reconnection in PT56.4S
```

--- a/integration-tests/src/test/java/com/datastax/oss/driver/core/metadata/DefaultMetadataTabletMapIT.java
+++ b/integration-tests/src/test/java/com/datastax/oss/driver/core/metadata/DefaultMetadataTabletMapIT.java
@@ -232,7 +232,7 @@ public class DefaultMetadataTabletMapIT {
           // Preparation of the statements without KS will fail on the session with no ks specified
           continue;
         }
-        CqlSession session = sessionEntry.getValue().get();
+        try (CqlSession session = sessionEntry.getValue().get()) {
         // Empty out tablets information
         if (session.getMetadata().getTabletMap().isPresent()) {
           session
@@ -281,6 +281,7 @@ public class DefaultMetadataTabletMapIT {
                   "Statement %s on session %s was routed to different nodes",
                   stmtEntry.getKey(), sessionEntry.getKey()));
         }
+        }
       }
     }
 
@@ -293,9 +294,9 @@ public class DefaultMetadataTabletMapIT {
 
   @Test
   public void should_receive_each_tablet_exactly_once() {
-    CqlSession session =
-        CqlSession.builder().addContactEndPoints(CCM_RULE.getContactPoints()).build();
     int counter = 0;
+    try (CqlSession session =
+      CqlSession.builder().addContactEndPoints(CCM_RULE.getContactPoints()).build()) {
     PreparedStatement preparedStatement = session.prepare(STMT_INSERT);
     for (int i = 1; i <= QUERIES; i++) {
       if (executeAndReturnIfResultHasTabletsInfo(session, preparedStatement.bind(i, i))) {
@@ -304,11 +305,11 @@ public class DefaultMetadataTabletMapIT {
     }
     Assert.assertEquals(INITIAL_TABLETS, counter);
     assertSessionTabletMapIsFilled(session);
-    session.close();
+    }
 
-    session = CqlSession.builder().addContactEndPoints(CCM_RULE.getContactPoints()).build();
+    try (CqlSession session = CqlSession.builder().addContactEndPoints(CCM_RULE.getContactPoints()).build()) {
     counter = 0;
-    preparedStatement = session.prepare(STMT_SELECT);
+    PreparedStatement preparedStatement = session.prepare(STMT_SELECT);
     for (int i = 1; i <= QUERIES; i++) {
       if (executeAndReturnIfResultHasTabletsInfo(session, preparedStatement.bind(i, i))) {
         counter++;
@@ -333,6 +334,7 @@ public class DefaultMetadataTabletMapIT {
         throw new RuntimeException(
             "Received non empty payload with tablets routing information: " + payload);
       }
+    }
     }
   }
 

--- a/integration-tests/src/test/java/com/datastax/oss/driver/core/metadata/DefaultMetadataTabletMapIT.java
+++ b/integration-tests/src/test/java/com/datastax/oss/driver/core/metadata/DefaultMetadataTabletMapIT.java
@@ -233,54 +233,54 @@ public class DefaultMetadataTabletMapIT {
           continue;
         }
         try (CqlSession session = sessionEntry.getValue().get()) {
-        // Empty out tablets information
-        if (session.getMetadata().getTabletMap().isPresent()) {
-          session
-              .getMetadata()
-              .getTabletMap()
-              .get()
-              .removeByKeyspace(CqlIdentifier.fromCql(KEYSPACE_NAME));
-        }
-        Statement stmt;
-        try {
-          stmt = stmtEntry.getValue().apply(session);
-        } catch (Exception e) {
-          RuntimeException ex =
-              new RuntimeException(
+          // Empty out tablets information
+          if (session.getMetadata().getTabletMap().isPresent()) {
+            session
+                .getMetadata()
+                .getTabletMap()
+                .get()
+                .removeByKeyspace(CqlIdentifier.fromCql(KEYSPACE_NAME));
+          }
+          Statement stmt;
+          try {
+            stmt = stmtEntry.getValue().apply(session);
+          } catch (Exception e) {
+            RuntimeException ex =
+                new RuntimeException(
+                    String.format(
+                        "Failed to build statement %s on session %s",
+                        stmtEntry.getKey(), sessionEntry.getKey()));
+            ex.addSuppressed(e);
+            throw ex;
+          }
+          try {
+            if (!executeOnAllHostsAndReturnIfResultHasTabletsInfo(session, stmt)) {
+              testErrors.add(
                   String.format(
-                      "Failed to build statement %s on session %s",
+                      "Statement %s on session %s got no tablet info",
                       stmtEntry.getKey(), sessionEntry.getKey()));
-          ex.addSuppressed(e);
-          throw ex;
-        }
-        try {
-          if (!executeOnAllHostsAndReturnIfResultHasTabletsInfo(session, stmt)) {
+              continue;
+            }
+          } catch (Exception e) {
             testErrors.add(
                 String.format(
-                    "Statement %s on session %s got no tablet info",
+                    "Failed to execute statement %s on session %s: %s",
+                    stmtEntry.getKey(), sessionEntry.getKey(), e));
+            continue;
+          }
+          if (!waitSessionLearnedTabletInfo(session)) {
+            testErrors.add(
+                String.format(
+                    "Statement %s on session %s did not trigger session tablets update",
                     stmtEntry.getKey(), sessionEntry.getKey()));
             continue;
           }
-        } catch (Exception e) {
-          testErrors.add(
-              String.format(
-                  "Failed to execute statement %s on session %s: %s",
-                  stmtEntry.getKey(), sessionEntry.getKey(), e));
-          continue;
-        }
-        if (!waitSessionLearnedTabletInfo(session)) {
-          testErrors.add(
-              String.format(
-                  "Statement %s on session %s did not trigger session tablets update",
-                  stmtEntry.getKey(), sessionEntry.getKey()));
-          continue;
-        }
-        if (!checkIfRoutedProperly(session, stmt)) {
-          testErrors.add(
-              String.format(
-                  "Statement %s on session %s was routed to different nodes",
-                  stmtEntry.getKey(), sessionEntry.getKey()));
-        }
+          if (!checkIfRoutedProperly(session, stmt)) {
+            testErrors.add(
+                String.format(
+                    "Statement %s on session %s was routed to different nodes",
+                    stmtEntry.getKey(), sessionEntry.getKey()));
+          }
         }
       }
     }
@@ -296,45 +296,46 @@ public class DefaultMetadataTabletMapIT {
   public void should_receive_each_tablet_exactly_once() {
     int counter = 0;
     try (CqlSession session =
-      CqlSession.builder().addContactEndPoints(CCM_RULE.getContactPoints()).build()) {
-    PreparedStatement preparedStatement = session.prepare(STMT_INSERT);
-    for (int i = 1; i <= QUERIES; i++) {
-      if (executeAndReturnIfResultHasTabletsInfo(session, preparedStatement.bind(i, i))) {
-        counter++;
+        CqlSession.builder().addContactEndPoints(CCM_RULE.getContactPoints()).build()) {
+      PreparedStatement preparedStatement = session.prepare(STMT_INSERT);
+      for (int i = 1; i <= QUERIES; i++) {
+        if (executeAndReturnIfResultHasTabletsInfo(session, preparedStatement.bind(i, i))) {
+          counter++;
+        }
       }
-    }
-    Assert.assertEquals(INITIAL_TABLETS, counter);
-    assertSessionTabletMapIsFilled(session);
+      Assert.assertEquals(INITIAL_TABLETS, counter);
+      assertSessionTabletMapIsFilled(session);
     }
 
-    try (CqlSession session = CqlSession.builder().addContactEndPoints(CCM_RULE.getContactPoints()).build()) {
-    counter = 0;
-    PreparedStatement preparedStatement = session.prepare(STMT_SELECT);
-    for (int i = 1; i <= QUERIES; i++) {
-      if (executeAndReturnIfResultHasTabletsInfo(session, preparedStatement.bind(i, i))) {
-        counter++;
+    try (CqlSession session =
+        CqlSession.builder().addContactEndPoints(CCM_RULE.getContactPoints()).build()) {
+      counter = 0;
+      PreparedStatement preparedStatement = session.prepare(STMT_SELECT);
+      for (int i = 1; i <= QUERIES; i++) {
+        if (executeAndReturnIfResultHasTabletsInfo(session, preparedStatement.bind(i, i))) {
+          counter++;
+        }
       }
-    }
 
-    LOG.debug("Ran first set of queries");
+      LOG.debug("Ran first set of queries");
 
-    // With enough queries we should hit a wrong node for each tablet exactly once.
-    Assert.assertEquals(INITIAL_TABLETS, counter);
-    assertSessionTabletMapIsFilled(session);
+      // With enough queries we should hit a wrong node for each tablet exactly once.
+      Assert.assertEquals(INITIAL_TABLETS, counter);
+      assertSessionTabletMapIsFilled(session);
 
-    // All tablet information should be available by now (unless for some reason cluster did sth on
-    // its own)
-    // We should not receive any tablet payloads now, since they are sent only on mismatch.
-    for (int i = 1; i <= QUERIES; i++) {
+      // All tablet information should be available by now (unless for some reason cluster did sth
+      // on its own). We should not receive any tablet payloads now, since they are sent only on
+      // mismatch.
+      for (int i = 1; i <= QUERIES; i++) {
 
-      ResultSet rs = session.execute(preparedStatement.bind(i, i));
-      Map<String, ByteBuffer> payload = rs.getExecutionInfo().getIncomingPayload();
+        ResultSet rs = session.execute(preparedStatement.bind(i, i));
+        Map<String, ByteBuffer> payload = rs.getExecutionInfo().getIncomingPayload();
 
-      if (payload.containsKey(TabletInfo.TABLETS_ROUTING_V1_CUSTOM_PAYLOAD_KEY)) {
-        throw new RuntimeException(
-            "Received non empty payload with tablets routing information: " + payload);
+        if (payload.containsKey(TabletInfo.TABLETS_ROUTING_V1_CUSTOM_PAYLOAD_KEY)) {
+          throw new RuntimeException(
+              "Received non empty payload with tablets routing information: " + payload);
+        }
       }
-    }
     }
   }
 

